compiler/rustc_trait_selection/src/traits/coherence.rs

```diff
@@ -132,7 +132,11 @@ fn with_fresh_ty_vars<'cx, 'tcx>(
         impl_def_id,
         self_ty: tcx.type_of(impl_def_id).subst(tcx, impl_substs),
         trait_ref: tcx.impl_trait_ref(impl_def_id).map(|i| i.subst(tcx, impl_substs)),
-        predicates: tcx.predicates_of(impl_def_id).instantiate(tcx, impl_substs).predicates,
+        predicates: tcx
+            .predicates_of(impl_def_id)
+            .instantiate(tcx, impl_substs)
+            .predicates
+            .to_vec(),
```

Comment:
Here and lower: `to_vec` clones into new vec, maybe `into_vec` should be used?

Reply:
Ah yes, in these cases `.into_vec()` should definitely be preferred.